php-checks/src/main/java/org/sonar/php/checks/AllBranchesIdenticalCheck.java

```diff
@@ -0,0 +1,130 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.AbstractDuplicateBranchCheck;
+import org.sonar.php.checks.utils.Equality;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.expression.ConditionalExpressionTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.ElseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.ElseifClauseTree;
+import org.sonar.plugins.php.api.tree.statement.IfStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.tree.statement.SwitchCaseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.SwitchStatementTree;
+
+@Rule(key = "S3923")
+public class AllBranchesIdenticalCheck extends AbstractDuplicateBranchCheck {
+
+  private static final String MESSAGE = "Remove this conditional structure or edit its code blocks so that they're not all the same.";
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpressionTree tree) {
+    if (Equality.areSyntacticallyEquivalent(tree.trueExpression(), tree.falseExpression())) {
+      context().newIssue(this, tree.condition(), tree.queryToken(), MESSAGE);
+    }
+    super.visitConditionalExpression(tree);
+  }
+
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    if (!checkedIfStatements.contains(tree)) {
+      List<List<StatementTree>> branches = new ArrayList<>();
+      boolean hasElse = false;
+
+      for (Tree clause : getClauses(tree)) {
+        if (clause.is(Kind.IF_STATEMENT)) {
+          IfStatementTree ifStatementTree = (IfStatementTree) clause;
+          branches.add(ifStatementTree.statements());
+
+        } else if (clause.is(Kind.ELSEIF_CLAUSE)) {
+          ElseifClauseTree elseifClauseTree = (ElseifClauseTree) clause;
+          branches.add(elseifClauseTree.statements());
+
+        } else if (clause.is(Kind.ELSE_CLAUSE)) {
+          ElseClauseTree elseClause = (ElseClauseTree) clause;
+          if (!elseClause.statements().get(0).is(Kind.IF_STATEMENT)) {
+            branches.add(elseClause.statements());
+            hasElse = true;
+          }
+        }
+      }
+
+      if (hasElse) {
+        checkBranches(branches, tree.ifToken());
+      }
+    }
+
+    super.visitIfStatement(tree);
+  }
+
+  @Override
+  public void visitSwitchStatement(SwitchStatementTree tree) {
+    boolean hasFallthrough = false;
+
+    List<List<StatementTree>> normalizedBranches = new ArrayList<>();
+    boolean hasDefault = false;
+
+    for (int i = 0; i < tree.cases().size(); i++) {
+      SwitchCaseClauseTree switchCaseClause = tree.cases().get(i);
+      boolean isLast = (i == tree.cases().size() - 1);
```

Comment:
I would go for something like : 
```
int lastIndex = tree.cases().sizes() -1
 for (int i = 0; i <= lastIndex; i++) {
    boolean isLast = (i == lastIndex);
```
and even eventually inline the `isLast` boolean as it is used only once.